### DISCUSSION TITLE
Search plans by location/careteam

### DISCUSSION THREAD

**Discussion by rkareko** [opensrp/fhircore#318] [Ideas]
How do we search for plans using a location or a care team

@rkodev @ekigamba @f-odhiambo @dubdabasoduba @ndegwamartin

**Comment by ekigamba** [comment]
Notes from the June 25th meeting:

- The PlanDefinition does not have a reference to a CareTeam or Location(Jurisdiction) and thus cannot be easily searched. The PlanDefinition has a jurisdiction property that is a country or location-code (CodeableConcept). However, the CarePlan has a reference to a PlanDefinition and CareTeam. Therefore, a CarePlan can potentially be used to tag the scope of the PlanDefinition i.e. CareTeam, Location etc

**Comment by f-odhiambo** [comment]
Check this out https://docs.google.com/drawings/d/1Fg2YeealVMRrRY9tls0BHhF01dScqgfaH280O7l_r9U

cc @rkodev @ekigamba @f-odhiambo @dubdabasoduba @ndegwamartin @Rkareko

**Comment by pld** [comment]
Do we need to this for PlanDefinitions or CarePlans?

1) are we trying to understand what PlanDefinitions are appropriate for a Location and CareTeam?
  - I don't think we have a good solution for that, and would need to investigate further

2) or is this after a PlanDefinition has been run to create a CarePlan for a location
  - Like in the diagram Francis shared, these will be tied to a location via the Group or Patient the CarePlan is for and via the Location tied to the Tasks tied to the CarePlan. CareTeam is only connected through the Group, is that a problem?

**Reply by ekigamba** [reply]
For context, if we plan to migrate Reveal to FHIR, then we need to maintain the current workflow. 

When a user installs Reveal and logs in, they have to chose an operational area and then chose a `PlanDefinition` available within that area

![Screenshot from 2021-07-01 19-46-50](https://user-images.githubusercontent.com/31766075/124160728-3c3cab00-daa5-11eb-998a-5eac6785104f.png)

The plans on Reveal are defined as `PlanDefinitions`

On your questions:
1. This is correct, we might need to investigate further
2. We can probably use a CarePlan for a location instead of using a PlanDefinition. It's probably not completely clear whether we would want to use CarePlans for patients or operational areas

**Reply by pld** [reply]
Cool, thanks, very helpful. Thoughts:

- Although PlanDefinitions don't have Location, ActivityDefinitions do
- The subject field of PlanDefinition can link to a Group which could define our "Operational Area"
- As you noted, if we define "Operational Area" as a codeable concept, we can link PlanDefinition jurisdictions to it, and since a PlanDefinition can link to many jurisdictions we can use this to build a many to many relationship

**Reply by ekigamba** [reply]
Thanks, I think we can consider the options above while reviewing this again.

I am not completely sure about using the jurisdiction field since the available value-set for this field seems to predefined country, country subdivision and region based on the ISO 3166 https://www.hl7.org/fhir/valueset-jurisdiction.html. Do you have an idea how we can extend this for our use-case?

**Reply by rkodev** [reply]
Using the PlanDefintion field will be a workable solution, point to note, PlanDefinition.subject is not a defined search parameter and therefore is currently only stored in JSON format in the database, we can explore the idea of allowing the SDK to generate indexes from fields not defined in the [FHIR Spec](https://www.hl7.org/fhir/plandefinition.html#search)
